src/point_traits.h

```diff
@@ -0,0 +1,56 @@
+#ifndef CATA_SRC_POINT_TRAITS_H
+#define CATA_SRC_POINT_TRAITS_H
+
+#include <type_traits>
+
+struct point;
+struct tripoint;
+
+template<typename Point, typename = void>
+struct point_traits {
+    static inline int &x( Point &p ) {
+        return p.x();
+    }
+    static inline int x( const Point &p ) {
+        return p.x();
+    }
+    static inline int &y( Point &p ) {
+        return p.y();
+    }
+    static inline int y( const Point &p ) {
+        return p.y();
+    }
+    static inline int &z( Point &p ) {
+        return p.z();
+    }
+    static inline int z( const Point &p ) {
+        return p.z();
+    }
+};
+
+template<typename Point>
+struct point_traits <
+    Point,
+    std::enable_if_t < std::is_same<Point, point>::value || std::is_same<Point, tripoint>::value >
+    >  {
+    static inline int &x( Point &p ) {
```

Comment:
```suggestion
    static int &x( Point &p ) {
```
A function defined within the class body is automatically inline. Applies to all the other inline function here.